lib/elixir/lib/stream.ex

```diff
@@ -197,6 +197,10 @@ defmodule Stream do
     {:cont, acc}
   end
 
+  defp do_chunk_by(acc({_, 1}, _, _) = acc, _f1) do
```

Comment:
Unfortunately this fix is not correct. We cannot access the accumulator state because the take value can be nested inside other states or it could just mean whatever. :)